File: Assignment/Abstraction/VariableNode.cs
namespace Assignment.Abstraction
{
    internal struct VariableNode: INode
    {
        public VariableNode(string value)
        {
            this.Value = value;
        }

        public string Value { get; }

        public string DisplayName()
        {
            return $"{nameof(VariableNode)}(value: {this.Value})";
        }
    }
}

File: src/main/java/compiler/abstraction/expressions/VariableNode.java
package compiler.abstraction.expressions;

import java.util.UUID;

import compiler.abstraction.INode;

public class VariableNode extends INode {
    public final String value;

    public VariableNode(String value) {
        super();
        this.value = value;
    }

    public VariableNode(String value, UUID id) {
        super(id);
        this.value = value;
    }

    @Override
    public String getDisplayName() {
        return String.format("%s(value: %s)", VariableNode.class.getSimpleName(), this.value);
    }
}